src/main/java/duke/command/Command.java

```diff
@@ -50,10 +49,10 @@ protected Command(String name, String description, String usage, Map<String, Str
 
     /**
      * Executes the command with parameters given by the user.
-     *  @param commandParams the parameters given by the user, parsed into a {@code CommandParams} object.
-     * @param expensesList  The ExpenseList of Duke.
-     * @param ui            The ui of Duke.
+     *
+     * @param commandParams the parameters given by the user, parsed into a {@code CommandParams} object.
+     * @param duke          The Duke object.
      */
-    public void execute(CommandParams commandParams, ExpenseList expensesList, Ui ui) throws DukeException {
+    public void execute(CommandParams commandParams, Duke duke) throws DukeException {
```

Comment:
this is fine, although if this is the case then we should have the main controller call commands instead of duke e.g. move getResponse into MainWindow